basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/LTIExportServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+import org.sakaiproject.lti.api.LTIService;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.util.FormattedText;
+import org.sakaiproject.util.ResourceLoader;
+
+public class LTIExportServiceImpl implements LTIExportService {
+	protected LTIService ltiService;
+	protected ServerConfigurationService serverConfigurationService;
+	protected SiteService siteService;
+	
+	private String csvSeparator = null;
+	
+	/** Resource bundle using current language locale */
+	protected static ResourceLoader rb = new ResourceLoader("ltiservice");
+	
+	private static Log M_log = LogFactory.getLog(LTIExportServiceImpl.class);
+	
+	public void setLtiService(LTIService ltiService) {
+		this.ltiService = ltiService;
+	}
+	
+	public void setServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+		this.serverConfigurationService = serverConfigurationService;
+	}
+
+	public void setSiteService(SiteService siteService) {
+		this.siteService = siteService;
+	}
+
+	public void init() {
+		
+	}
+	
+	private boolean isAdmin(String siteId) {
+		if ( siteId == null ) {
+			throw new java.lang.RuntimeException("isAdmin() requires non-null siteId");
+		}
+		if (!"!admin".equals(siteId)) return false;
+		return siteService.allowUpdateSite(siteId);
+	}
+
+	/* (non-Javadoc)
+	 * @see impl.LTIExportService#export(java.io.OutputStream, int, java.lang.String)
+	 */
+	public void export(OutputStream out, String siteId, int exportType, String filterId) {
+		String search = null;
+		//check if we need to filter the tools by tool_id
+		if(StringUtils.isNotEmpty(filterId)) {
+			search = "tool_id:"+filterId;
+		}
+		List<Map<String,Object>> contents = ltiService.getContentsDao(search, null, 0, 0, siteId, isAdmin(siteId));
+		switch(exportType) {
+			case TYPE_CSV : exportCSV(out, siteId, contents); break;
+			case TYPE_EXCEL : exportExcel(out, siteId, contents); break;
+		}
+	}
+	
+	private String getCSVSeparator() {
+		if(csvSeparator == null) {
+			csvSeparator = serverConfigurationService.getString("basiclti.export.csv.separator", ",");
+		}
+		return csvSeparator;
+	}
+	
+	private void exportCSV(OutputStream out, String siteId, List<Map<String,Object>> contents) {
+		
+		StringBuilder result = new StringBuilder();
+		FormattedText validator = new FormattedText();
+		DateFormat dateFormatter = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT, rb.getLocale());
+		
+		String attribution_name = serverConfigurationService.getString(LTIService.LTI_SITE_ATTRIBUTION_PROPERTY_NAME);
+		if(StringUtils.isNotEmpty(attribution_name)) {
+			//check if given property is a translation key
+			String aux = rb.getString(attribution_name);
+			if(StringUtils.isNotEmpty(aux)) 
+				attribution_name = aux;
+		}
+		
+		boolean isAdmin = isAdmin(siteId);
+		
+		//set header row
+		result.append(rb.getString("export.title"));
+		result.append(getCSVSeparator());
```

Comment:
This will break if a site title has a comma in it. When building a CSV you should correctly escape all the values, normally you do this by quoting all values in double quotes. Double quotes should be escaped with a preceding double quote. Although it's not complex you could use a CSV library.